OpenSSL/crypto.py

```diff
@@ -2304,6 +2372,19 @@ def verify(cert, signature, data, digest):
         _raise_current_error()
 
 
+def verify_cert(store_ctx):
+    """
+    Verify a certificate in a context.
+
+    :param store_ctx: The :py:class:`X509StoreContext` to verify.
+    :raises: Error
+    """
+    store_ctx._init()
+    ret = _lib.X509_verify_cert(store_ctx._store_ctx)
+    store_ctx._cleanup()
```

Comment:
Thanks for pointing that out.  Sorry about the inconsistent feedback (I'm definitely having trouble remembering all of the discussions and suggestions made on this issue so far).

Fortunately it's all written down, hooray.  I went back and read some older comments.  One thing I recognized that somehow I'd overlooked recently is that the wrapped API is `X509_verify_cert`.  Going by the pyOpenSSL convention, that suggests this is actually a method of `X509`.  Too bad that doesn't actually make sense!  Because the only argument accepted is an `X509StoreContext`.  Argh.  I think OpenSSL just named the function wrong and they should have called it `X509_STORE_CTX_verify_cert`.  Alternatively, the X509 itself shouldn't be part of the X509_STORE_CTX and the signature should have been `X509_verify_cert(x509, x509_store_ctx)`.  But I suppose wishing the API and naming made more sense in OpenSSL doesn't help us at all.

I think I prefer the conclusion that `X509_verify_cert` is really an OpenSSL API that concerns itself first and foremost with `X509_STORE_CTX` structures (silly as that structure may be) and so the proper translation of this API into pyOpenSSL is as an `X509StoreContext` method.

There is precedent in pyOpenSSL for methods that are slightly more convenient than the (straightforward Python translation of the) OpenSSL APIs, though (eg `load_privatekey`).  So perhaps some good **follow-up** work would be a top-level pyOpenSSL `verify_cert` (or `verify_certificate` - I kinda dislike ad hoc abbreviations :smile:) that takes a handful of arguments, makes the `X509StoreContext` instance for you and calls its `verify_cert` method.  Most people will probably not care that `X509StoreContext` happens to be a thing (but exposing it is still a good idea for more advanced users).

Does that make sense and sound like a good way forward?